Feature/step3

river-o opened the pull request:
- 자세한 구현 사항과 기능은 [README.md](https://github.com/river-o/w2-signup-profile/tree/feature/step3) 에서 확인하실 수 있습니다.

river-o committed:
Init project

river-o committed:
feat: Tab Bar Controller 추가

river-o committed:
feat: UI Label 연결 및 세팅

river-o committed:
feat: Button Touch Event 생성

river-o committed:
feat: Edit 화면 UI 및 데이터 전달

- Edit 화면 UI
- 전 화면에서 데이터 전달

river-o committed:
feat: 이전 화면으로 돌아가는 기능 추가

river-o committed:
feat: 콜백 함수 호출

river-o committed:
feat: 프로필 사진 변경 기능

river-o committed:
feat: 프로필 내용 편집 기능

river-o committed:
docs: Update README.md

river-o committed:
feat: Navigation Controller 적용 및 Login 화면 UI 구성

river-o committed:
feat: 로그인 성공 시 Main 화면 이동

river-o committed:
feat: Main 화면에 WebView 띄우기

river-o committed:
docs: Update README.md

timothy-podo commented on KakaoProfile/KakaoProfile/LoginViewController.swift:
```diff
@@ -0,0 +1,58 @@
+        let idText = idEditText.text
+        let passwordText = passwordEditText.text
+        
+        if(idText == id && passwordText == password) {
+            self.performSegue(withIdentifier: onLoginSuccessSegue, sender: nil)
+        }
```
미션에는 없었을거 같기는 하지만, ID, PW 가 다를 때 알림창을 통해서 사용자에게 계정정보가 틀렸다고 알려주면 더욱 좋을거 같아요~ :)

river-o replied:
이 부분은 제가 생각하지 못했네요!~ alert 창을 통해 계정정보가 틀렸다는 정보를 주도록 수정하겠습니다. 피드백 감사합니다!! :)

timothy-podo commented on KakaoProfile/KakaoProfile/MainViewController.swift:
```diff
@@ -0,0 +1,66 @@
+    
+    private func initWebView() {
+        webView.uiDelegate = self
+        view = webView
```
뷰 컨트롤러의 뷰를 교체하고 싶으신 거면 loadView 를 override 해서 해 주시는게 좋을거 같아요.
관련 애플 레퍼런스 링크 공유 드리니 참고부탁드릴게요~ :)
https://developer.apple.com/documentation/uikit/uiviewcontroller/1621454-loadview

문서에 이런 코멘트가 있는데요.
>If you use Interface Builder to create your views and initialize the view controller, you must not override this method.

혹시 스토리보드로 생성한 ViewController 라면 view 를 webView 로 교체하는 방법보다, webView 를 view 의 chid 로 add 하는 방법도 있어서요. 이 문서도 추가로 공유 드리오니 참고해 주시면 좋겠습니다~ :)
https://developer.apple.com/documentation/uikit/uiview/1622616-addsubview/

river-o replied:
스토리보드로 생성한 ViewController에서는 loadView()를 override 하면 안 되는군요..! 
스토리보드로 생성한 ViewController라 말씀해주신 대로 webview를 main view의 child로 add 하는 방식으로 변경했습니다! 피드백 감사합니다! :)
756a212

river-o committed:
feat: 상수 enum 처리

river-o committed:
feat: Login TextField에 Clear Button 추가

river-o committed:
feat: 로그인 실패 시 알림창 띄워주는 기능

kyoo-sl commented on KakaoProfile/KakaoProfile/EditProfileDataProtocol.swift:
```diff
@@ -0,0 +1,13 @@
+import Foundation
+import UIKit
+
+protocol EditProfileDataProtocol: class {
```
저는 아래와 같은 워닝이 뜨는데, @river-o도 엑스코드에서 돌리면 아래 워닝이 뜨나요??
`Using 'class' keyword to define a class-constrained protocol is deprecated; use 'AnyObject' instead`

river-o replied:
넵..! 저도 다음과 같은 워닝이 발생합니다. 미처 발견하지 못했었네요! 
워닝 문구대로 class는 deprecated 되었기에 AnyObject 를 사용하는 방식으로 고쳤습니다! 피드백 감사합니다 :)
c0373c8

kyoo-sl commented on KakaoProfile/KakaoProfile/ProfileViewController.swift:
```diff
@@ -0,0 +1,100 @@
+        descriptionLabel.text = descriptionText
+        
+        nameLabel.font = UIFont.systemFont(ofSize: 24)
+        descriptionLabel.font = UIFont.systemFont(ofSize: 15)
```
setUI는 프로필을 edit 한 후 매번 호출되는데요, 
font처럼 초기화할 때 한번한 해도 되는 설정들은 initUI로 옮기면 좋을 것 같아요~

river-o replied:
말씀해주신 대로 한 번만 해도 되는 설정들은 initUI로 옮겼습니다! 조금 더 효율적으로 코드를 사용할 수 있게 되었습니다!! 감사합니다!! :) 
c0373c8

river-o committed:
feat: Main 화면의 Child View로 WebView 추가

river-o committed:
feat: PR 피드백 반영

- class -> AnyObject로 변경
- ProfileViewController에서 한 번만 설정해도 되는 세팅들 initUI() 함수로 이동

kyoo-sl reviewed (approved):
확인했습니다~
